libs/payments/capability/src/lib/capability.manager.ts

```diff
@@ -40,14 +40,14 @@ export class CapabilityManager {
   ): Promise<Record<string, string[]>> {
     if (!subscribedPrices.length) return {};
 
-    const purchaseDetails =
-      await this.contentfulManager.getPurchaseDetailsForCapabilityServiceByPlanIds(
-        [...subscribedPrices]
-      );
-
     const result: Record<string, string[]> = {};
 
     for (const subscribedPrice of subscribedPrices) {
+      const purchaseDetails =
+        await this.contentfulManager.getPurchaseDetailsForCapabilityServiceByPlanIds(
+          [subscribedPrice]
+        );
+
```

Comment:
Why is this change necessary? Is it what we discussed previously to limit the number of possible cacheKeys to the number of plans?

Reply:
Correct - it's to satisfy bullet point 2 in the ticket:

> Call getPurchaseDetailsForCapabilityServiceByPlanIds for plan IDs individually, such that we greatly increase the likelihood of a cache hit